components/quick_switch_modal/index.js

```diff
@@ -3,21 +3,25 @@
 
 import {connect} from 'react-redux';
 
-import {goToChannel, goToChannelById, openDirectChannelToUser} from 'actions/channel_actions.jsx';
+import {goToChannel, openDirectChannelToUser} from 'actions/channel_actions.jsx';
+
+import {makeGetChannelUrlById} from 'utils/channel_utils';
 
 import QuickSwitchModal from './quick_switch_modal.jsx';
 
-function mapStateToProps() {
+function mapStateToProps(state) {
+    const getChannelUrlById = makeGetChannelUrlById(state);
```

Comment:
This doesn't work since it's returning a new function each time. This would be a bit better, but it will sometimes return a new function when it doesn't really matter
```javascript
export const makeGetChannelUrlById = createSelector(
    getChannels,
    getCurrentTeamRelativeUrl,
    (channels, teamUrl) => {
        // TODO get channel and return url
    }
);
```

That being said, this request might contradict Saturn's previous one, but I'd prefer to keep a `goToChannelById` action like the old one. The old version was very close to what we'd want, but it would need to be modified slightly to be properly bound to to the store
```javascript
export function goToChannelById(channelId) {
    return (dispatch, getState) => {
        const channel = getChannel(getState(), channelId);
        browserHistory.push(TeamStore.getCurrentTeamRelativeUrl() + '/channels/' + channel.name);
    };
}
```
While that makes the action itself harder to test since you need to mock the browserHistory there (maybe that actually can be passed into the action like the client in the changes I was discussing earlier?), you have to mock the browserHistory when testing both components now.